addons/mail/static/src/core/web/activity_button.js

```diff
@@ -45,8 +45,9 @@ export class ActivityButton extends Component {
                 classes.push(this.props.record.data.activity_exception_icon);
                 break;
             default:
-                if (this.props.record.data.activity_type_icon) {
-                    classes.push(this.props.record.data.activity_type_icon);
+                const { activity_ids, activity_type_icon } = this.props.record.data;
+                if (activity_ids.records.length) {
+                    classes.push(activity_type_icon || "fa-tasks btn-link text-dark");
```

Comment:
1. Missing scope inside a `switch/case` branching, which is required by linter to define `const`
2. I would remove `btn-link text-dark`, as this is style is reserved for "no activities". This would make color match the "overdue"/"today"/"planned" status. Yes, the icon visual will be the same as `To-Do`, which is intentional with this PR